### PR TITLE
Add parameter to modify value of now

### PR DESCRIPTION
**Pull request by gamebox**
Add a parameter that allows for overriding the value of `now` to be a dynamic value
useful for syncing with server time if reliance on user clock not advisable.  Add unit
tests and fix current unit test problems.

Closes Issue #15

### CODE REVIEW

File: angular-relative-date.js [hunk 1]
```diff
@@ -60,7 +61,7 @@
         year = day * 365;
         delta = calculateDelta(now, date);
         if (delta > day && delta < week) {
-          date = new Date(date.getFullYear(), date.getMonth(), date.getDate(), 0, 0, 0);
+          date = new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate(), 0, 0, 0);
```
Comment (wildlyinaccurate) [inline angular-relative-date.js]: I'm just trying to figure out the implications of this change. Won't this result in incorrect deltas for users outside of UTC? Perhaps I need to extend the test suite to verify this.

File: test/spec/relative-date-filter.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ describe('Filter: relativeDate', function() {
     expect(relativeDate('2013-09-06T12:00:00Z')).toEqual('yesterday')
     expect(relativeDate('2013-09-06T00:00:00Z')).toEqual('yesterday')
     expect(relativeDate('2013-09-05T22:59:59Z')).toEqual('2 days ago')
-    expect(relativeDate('2013-09-01')).toEqual('6 days ago')
+    expect(relativeDate('2013-09-01T12:00:00Z')).toEqual('6 days ago')
```
Comment (wildlyinaccurate) [inline test/spec/relative-date-filter.js]: This worries me a little -- is this because of the `getDate` -> `getUTCDate` change?

**Comment by wildlyinaccurate** [comment]
When I have a bit more time (probably this weekend) I'll have a proper think about this change to make sure we don't break backwards compatibility -- especially for users in other timezones.

Apologies for the nitpicking but would you mind following the existing style in `test/spec/relative-date-filter.js` and remove the semicolons?

File: test/spec/relative-date-filter.js (hunk 2 above)
Comment (gamebox) [inline test/spec/relative-date-filter.js]: I don't quite remember, but I think I was getting failed tests before I made this change.  Will look into it this weekend

File: angular-relative-date.js (hunk 1 above)
Comment (gamebox) [inline angular-relative-date.js]: Maybe do the same for now here as well?  Just want to ensure that we are dealing with the same time zone.  I'll go back and look at this this weekend.  I only spent 10-15 minutes on this PR, just wanted you to get a full sense of what was involved.

**Comment by gamebox** [comment]
Sorry, I am not a CoffeeScript guy, so semicolons are life for me.  Plus, not having them can sometimes lead to unexpected behavior in vanilla JS.  But I will definitely follow your conventions.

**Pull request closed without merging**